syntax = "proto3";

message RegistrationDetails {
    string first_name = 2;
	string last_name = 3;
	string email = 4;
	bytes photo = 5;
	string photo_content_type = 6;
	string password = 7;
	repeated string roles = 8;
}

message UpdateDetails {
    string first_name = 2;
	string last_name = 3;
	string email = 4;
	bytes photo = 5;
	string photo_content_type = 6;
	repeated string added_roles = 7;
	repeated string removed_roles = 8;
}

message PasswordResetRequest {
   string email = 1;
}

message PasswordReset {
	string reference = 1;
	string otp = 2;
	string password = 3;
}